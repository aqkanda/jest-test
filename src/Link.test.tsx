import renderer from "react-test-renderer";
import Link from "./Link";

// スナップショットテスト
//スナップショットのUIと違うUIになった場合にFAILにする
it("hover時のクラス変更", () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );
  //通常時のスナップショットを作成
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  //ホバー時のスナップショットを作成
  //hooksを使用しているときはactを使う
  renderer.act(() => {
    //@ts-ignore
    tree.props.onMouseEnter();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // ホバー解除時のスナップショットを作成
  renderer.act(() => {
    //@ts-ignore
    tree.props.onMouseLeave();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
