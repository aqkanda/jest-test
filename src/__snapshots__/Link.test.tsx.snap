// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hover時のクラス変更 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`hover時のクラス変更 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`hover時のクラス変更 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
